### PR TITLE
opentelemetry: refactor span links code for clarity

### PR DESCRIPTION
**Pull request by katiehockman**
### What does this PR do?

This change also includes a number of refactors, including some additional information about the trace flags.

### Motivation

We should maintain some of the logic in the original PR when creating the DD span link: https://github.com/DataDog/dd-trace-go/pull/2396/files#diff-328a045c801a15647d453336c495d31dcd8016d29df86de478c77769d6792567

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddtrace/opentelemetry/span_test.go [hunk 1]
```diff
@@ -135,59 +135,54 @@ func TestSpanSetName(t *testing.T) {
...
-	assert.Equal(spanLinks[0].SpanID, uint64(15))
-	assert.Equal(spanLinks[0].Attributes, map[string]string{"link.name": "alpha_transaction"})
-	assert.Equal(spanLinks[0].Tracestate, "dd_origin=ci")
-	assert.Equal(spanLinks[0].Flags, uint32(0x80000000))
+	assert.Equal(uint64(123), spanLinks[0].TraceID)
+	assert.Equal(uint64(456), spanLinks[0].TraceIDHigh)
+	assert.Equal(uint64(15), spanLinks[0].SpanID)
+	assert.Equal(map[string]string{"link.name": "alpha_transaction"}, spanLinks[0].Attributes)
+	assert.Equal(traceStateVal, spanLinks[0].Tracestate)
+	assert.Equal(uint32(0x80000001), spanLinks[0].Flags) // sampled and set
```
Comment (mabdinur) [inline ddtrace/opentelemetry/span_test.go:185]: This assertion might fail. spanLinks[0].Flags is serialized to float64 -> json bytes -> uint32. The conversion to float64 might introduce precision issues.

**Review by mabdinur — APPROVED** [review]
Left one question. Looks good to me! Thanks for catching these issues

File: ddtrace/opentelemetry/span_test.go (hunk 1 above)
Comment (katiehockman) [reply ddtrace/opentelemetry/span_test.go:185]: I'm not sure I understand. Are you talking about a possible bug in the code, or just a possible issue with the test? I can at least confirm that this test passes as is without issue.

Comment (mabdinur) [reply ddtrace/opentelemetry/span_test.go:185]: Issue in the test. If it passes then this isn't an issue.

Comment (katiehockman) [reply ddtrace/opentelemetry/span_test.go:185]: It does indeed pass 😄

**Review by darccio — APPROVED** [review]

**Pull request merged**